pkgs/top-level/all-packages.nix

```diff
@@ -12741,6 +12741,7 @@ with pkgs;
 
   inherit (callPackages ../servers/teleport {
     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security AppKit;
+    buildGoModule = buildGo122Module;
```

Comment:
Realistically we could do `if stdenv.hostPlatform.isx86_64 then buildGo122Module else buildGoModule` but I don't think it's worth using different Go compiler versions on different arches.

Reply:
> I don't think it's worth using different Go compiler versions on different arches.

Agreed, that would complicate the build for very little gain